src/parser/Parser.ts

```diff
@@ -1235,10 +1230,24 @@ export class Parser {
         });
     }
 
-    private exitWhile(): ExitWhileStatement {
-        let keyword = this.advance();
+    private exitStatement(): ExitStatement {
+        const exitToken = this.advance();
+        if (exitToken.kind === TokenKind.ExitWhile) {
+            // `exitwhile` is allowed in code, and means `exit while`
+            // use an ExitStatement that is nicer to work with
 
-        return new ExitWhileStatement({ exitWhile: keyword });
+            return new ExitStatement({
+                exit: createToken(TokenKind.Exit, 'exit', exitToken.location),
+                loopType: createToken(TokenKind.While, 'while', exitToken.location)
```

Comment:
We need to more accurately represent the tokens as they were originally defined. So that means grabbing the raw text from the tokens instead of hardcoding `'exit'` and `'while'`, as well as their locations. Something like

```
exit: createToken(TokenKind.Exit, exitToken.substring(0, 4), /* first 4 chars of the range */),
loopType: createToken(TokenKind.While, exitToken.substring(4, 9), /* last 5 chars of the range */)